### ISSUE TITLE
Image Loader / Webpack (SVG) ?

### ISSUE THREAD

**Issue by mtompkins** [chentsulin/electron-react-boilerplate#980]
Apologies if this is self-evident and I've missed it, 
I'm still learning the ins-and-outs of this boilerplate (and React in general).

In short, I'm trying to add images to a [toast](https://github.com/fkhadra/react-toastify) component. 

With `Foo.js` in `app/components` and import with:
`import logo from '../img/logo.svg';` 
the project runs / builds / packages as expected and the reference of `src={logo}` works.

I've created an `app/js/helpers` folder.

With `Bar.js` in `app/js/helpers`and import with:
`import logo from '../../img/logo.svg';`
everything fails with (snipped):

```
App threw an error during load
SyntaxError: /home/mtompkins/Electron/react-redux/r3/app/img/logo.svg: Unexpected token (1:1)
> 1 | <?xml version="1.0" encoding="UTF-8" standalone="no"?>
    |  ^
  2 | <svg
  3 |    xmlns:dc="http://purl.org/dc/elements/1.1/"
  4 |    xmlns:cc="http://creativecommons.org/ns#"
    at Parser.pp$5.raise (/home/mtompkins/Electron/react-redux/r3/node_modules/babylon/lib/index.js:4443:13)
    at Parser.pp.unexpected (/home/mtompkins/Electron/react-redux/r3/node_modules/babylon/lib/index.js:1755:8)
    at Parser.pp$9.jsxParseIdentifier (/home/mtompkins/Electron/react-redux/r3/node_modules/babylon/lib/index.js:6887:10)
    at Parser.pp$9.jsxParseNamespacedName (/home/mtompkins/Electron/react-redux/r3/node_modules/babylon/lib/index.js:6898:19)
    at Parser.pp$9.jsxParseElementName (/home/mtompkins/Electron/react-redux/r3/node_modules/babylon/lib/index.js:6913:19)
    at Parser.pp$9.jsxParseOpeningElementAt (/home/mtompkins/Electron/react-redux/r3/node_modules/babylon/lib/index.js:7003:20)
. . . 
```

I'm not sure if this is an issue or something I'm just fundamentally missing.

Thx

**Comment by amilajack** [comment]
Try look into https://github.com/webpack-contrib/svg-inline-loader adding that config to your `webpack.base.config.js`

**Comment by mtompkins** [comment]
Hi @amilajack - I actually tried it, but only with a very quick effort waiting on some feedback to my inquiry. I'll give it a real look and see if it helps and revert.

Thanks for the recommendation.

**Comment by mtompkins** [comment]
Update: A few days in with little progress. I've put the following in hopes of some additional thoughts:
http://stackoverflow.com/questions/43896174/svg-webpack-loader-config-with-react-template-icon-loader

`svg-inline-loader` had a number of difficulties I didn't feel I was close to overcoming.

**Comment by amilajack** [comment]
What about this [svg-url-loader](https://github.com/bhovhannes/svg-url-loader)?

**Comment by mtompkins** [comment]
Small step forward... changing the loader from 'url-loader' to a 'file-loader' for SVG files has been productive. Once I have figured out how to properly incorporate it in to a formal Icon loader (which I believe to be a fundamental need in the template) I'll update and if you'd like offer a pull.

**Comment by amilajack** [comment]
A PR would be awesome!

**Comment by mtompkins** [comment]
So I've been able to determine that the problems aren't with any of the loaders, but the fundamental structure of your template as well as the associated bifurcation of webpack dev/prod. The svg assets simply aren't being included in the `dist` output and references in dev versus production aren't seamless. While it is great to know the origin of the problem it is also unfortunate that the fix isn't worth the effort of altering the structure for my use case.

At the end of the day, it's the choice of the coexistence of two structures causing the problem. 
Thanks anyway and good luck with your template - I'll have to move on to another direction.

**Comment by brandonmp** [comment]
i fiddled w/ this awhile ago, so not sure if this is still a valid solution for the current boilerplate, but adding this loader to `webpack.config.renderer.prod.js` worked for me to load svgs (via the `src` tag of an `<img>` element):


```
{
                test: /\.svg$/,
                loader: 'file-loader',
                options: {
                    // necessary to make this work w/ the output options in
                    // config.renderer.prod.js. w/o this all svgs path to ../dist
                    publicPath: './dist/'
                }
            }
```

which allows for

```
import SomeIcon from './some-icon.svg'

...
<img src={SomeIcon} />
...
```

**Comment by amilajack** [comment]
Yup. See https://github.com/chentsulin/electron-react-boilerplate/issues/1156#issuecomment-319300534. The `publicPath` in `v0.12.0` is wrong. Fixed in `v0.13.0`
